kernel/src/services.rs

```diff
@@ -66,6 +66,14 @@ pub struct SystemServices {
     pub servers: [Option<Server>; MAX_SERVER_COUNT],
 }
 
+pub enum PostActivateOp<'a> {
+    None,
+    // (result)
+    SetThreadResult(xous_kernel::Result),
+    // (sidx, current_pid, current_thread, message, client_address)
```

Comment:
It's actually possible to convert this to a named struct (the rust-analyzer API should allow you to refactor this, then hit F2 on the generated names to rename them)

Reply:
ooh huh since wen could you do this trick of named structs instead of enums? was it always a thing you can do interchangeably? this is so much better than anonymous enums, why would anon enums even exist when you can do this...